internal/service/loadbalancer/frontend_tls_config.go

```diff
@@ -2,153 +2,261 @@ package loadbalancer
+			"Unable to unmarshal loadbalancer frontend name",
+			utils.ErrorDiagnosticDetail(err),
+		)
+
+		return
 	}
 
-	if diags = setFrontendTLSConfigResourceData(d, t); len(diags) > 0 {
-		return diags
+	tlsConfig, err := r.client.GetLoadBalancerFrontendTLSConfig(ctx, &request.GetLoadBalancerFrontendTLSConfigRequest{
+		ServiceUUID:  loadbalancer,
+		FrontendName: frontendName,
+		Name:         data.Name.ValueString(),
+	})
+	if err != nil {
+		if utils.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				"Unable to read loadbalancer frontend TLS config details",
+				utils.ErrorDiagnosticDetail(err),
+			)
+		}
+		return
 	}
 
-	return diags
+	resp.Diagnostics.Append(setFrontendTLSConfigValues(ctx, &data, tlsConfig)...)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
-func resourceFrontendTLSConfigUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
-	svc := meta.(*service.Service)
-	var serviceID, feName, name string
-	if err := utils.UnmarshalID(d.Id(), &serviceID, &feName, &name); err != nil {
-		return diag.FromErr(err)
+func (r *frontendTLSConfigResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data frontendTLSConfigModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	var loadbalancer, frontendName string
+	err := utils.UnmarshalID(data.Frontend.ValueString(), &loadbalancer, &frontendName)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to unmarshal loadbalancer frontend name",
+			utils.ErrorDiagnosticDetail(err),
+		)
+
+		return
 	}
-	t, err := svc.ModifyLoadBalancerFrontendTLSConfig(ctx, &request.ModifyLoadBalancerFrontendTLSConfigRequest{
-		ServiceUUID:  serviceID,
-		FrontendName: feName,
-		Name:         name,
+
+	apiReq := &request.ModifyLoadBalancerFrontendTLSConfigRequest{
+		ServiceUUID:  loadbalancer,
+		FrontendName: frontendName,
+		Name:         data.Name.ValueString(),
 		Config: request.LoadBalancerFrontendTLSConfig{
-			Name:                  d.Get("name").(string),
-			CertificateBundleUUID: d.Get("certificate_bundle").(string),
+			Name:                  data.Name.ValueString(),
+			CertificateBundleUUID: data.CertificateBundle.ValueString(),
 		},
-	})
-	if err != nil {
-		return diag.FromErr(err)
 	}
 
-	d.SetId(utils.MarshalID(serviceID, feName, t.Name))
-
-	if diags = setFrontendTLSConfigResourceData(d, t); len(diags) > 0 {
-		return diags
+	tlsConfig, err := r.client.ModifyLoadBalancerFrontendTLSConfig(ctx, apiReq)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to modify loadbalancer frontend TLS config",
+			utils.ErrorDiagnosticDetail(err),
+		)
+		return
 	}
 
-	tflog.Info(ctx, "frontend TLS config updated", map[string]interface{}{"name": t.Name, "service_uuid": serviceID, "fe_name": feName})
-	return diags
+	resp.Diagnostics.Append(setFrontendTLSConfigValues(ctx, &data, tlsConfig)...)
```

Comment:
_(Same comment as in backend update)_ Is the ID now updated after rename like it was done in SDKv2 version?

Reply:
Thanks for noticing. The ID is now updated like it should be.